sdk/core/src/request.rs

```diff
@@ -0,0 +1,60 @@
+use http::{HeaderMap, Method};
+use std::fmt::Debug;
+use url::Url;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Request {
+    pub url: Url,
+    pub method: Method,
+    pub headers: HeaderMap,
+    pub payload: bytes::Bytes,
+}
+
+impl Request {
+    #[allow(dead_code)]
+    pub(crate) fn into_parts(self) -> (Parts, bytes::Bytes) {
+        let parts = Parts {
+            url: self.url,
+            method: self.method,
+            headers: self.headers,
+        };
+        (parts, self.payload)
+    }
+
+    pub fn set_payload(&mut self, bytes: bytes::Bytes) {
```

Comment:
Is the plan to eventually allow for non-bytes related payloads(e.g., streams)?

Reply:
Yes, it's something I was working on! Also I would like to update the response to better expose the return stream if need be.